src/rshim.c

```diff
@@ -2697,13 +2992,10 @@ static void rshim_main(int argc, char *argv[])
         rshim_pcie_lf_init_done = true;
       }
     } else {
-      /* Disable the timer if no rshim devices are found. */
-      if (rshim_dev_bitmask) {
-        if (!rshim_timer_interval)
-          rshim_set_timer(timer_fd, RSHIM_TIMER_INTERVAL);
-      } else if (rshim_timer_interval) {
-          rshim_set_timer(timer_fd, 0);
-      }
+      /* Always enable the timer even if no rshim devices are found, so we can
```

Comment:
could you please explain how this works. My understanding from the code:
1) Rshim USB on BMC is enabled
2) user enables rshim pcie on x86  host and uses FORCE CMD to take control over rshim.
3) The ownership transfer is done through writes of scratchpad1.
4) Now the user has no access from BMC rshim USB except for dumping misc. misc shows that it is in monitor mode (=1) and rshim access = 0. For rshim pcie, monitor mode =0 , force mode =1, rshim access = 1.

what matters to me is what happens after this:
1) the user has the ability to take back the BMC rshim USB by setting FORCE_CMD=0? 
2) and this is why we keep the misc file alive even if the rshim device is deregistered. So the only file we would see under /dev/rshimX/ is misc?

my question is why dont we just transfer the ownership completely to rshim pcie once we set FORCE_CMD=1? and deregister properly the rshim USB. if the user needs to take control again from rshim usb, they can just issue FORCE_CMD=1 from the rshim usb itself.

Reply:
* "FORCE_CMD" is a one-time command like "SW_RESET", so it can only be set to 1 and not 0.

* Apologize for didn't mention the RM number in description: https://redmine.mellanox.com/issues/3679848. You can see the use case for this feature is for BMC to take ownership from Host, kind of opposite to your example. In that case, the user will set "FORCE_CMD" to 1 to take ownership of RSHIM from host, and he/she cannot issue commands from the host for lack of access.

Reply:
In my new push below, I also improved the "FORCE_CMD" misc display to avoid the perception that "0" is allowed to write. Now it's similar to "SW_RESET".